data_dump/dump.py

```diff
@@ -0,0 +1,211 @@
+        ds = xr.Dataset(
+            {config.getKey(): (("report_step", "index"), data)},
+            coords={
+                "report_step": report_steps,
+                "index": range(data.shape[1]),
+            },
+        )
+        return "GEN_DATA", ds
+
+    def _load_gen_kw(
+        self, config: EnkfConfigNode, realization_index: int
+    ) -> Tuple[str, xr.Dataset]:
+        model = config.getKeywordModelConfig()
+        node = EnkfNode(config)
+        node.load(self.fs, NodeId(0, realization_index))
+        node = node.asGenKw()
+
+        data = [
+            (node._iget_key(i), node._data_iget(i, False), node._data_iget(i, True))
+            for i in range(len(node))
+        ]
+        names, raw, transformed = list(zip(*data))
+
+        funcs = [x["function"] for x in model.get_priors()]
+        params = [json.dumps(x["parameters"]) for x in model.get_priors()]
+
+        ds = xr.Dataset(
+            {
+                "standard_normal": ("name", list(raw)),
+                "transformed": ("name", list(transformed)),
+                "probability_distribution": ("name", funcs),
+                "probability_distribution_parameters": ("name", params),
+            },
+            coords={"name": ("name", list(names))},
+        )
+        return f"GEN_KW/{config.getKey()}", ds
+
+    def _load_surface(
+        self, config: EnkfConfigNode, realization_index: int
+    ) -> Tuple[str, xr.Dataset]:
+        node = EnkfNode(config)
+        node.load(self.fs, NodeId(0, realization_index))
+        _clib.surface.generate_parameter_file(node, "/tmp", "surface")
+        surface = xtgeo.surface_from_file("/tmp/surface", fformat="irap_ascii")
+        return f"SURFACE/{config.getKey()}", xr.Dataset(surface.dataframe())
+
+    def load_all(self) -> Iterable[Tuple[int, str, xr.Dataset]]:
+        ensemble_config = self.ert_config.ensemble_config
+        keys = ensemble_config.alloc_keylist()
+        for key in keys:
+            for realization_index in range(self.ert.getEnsembleSize()):
+                config = ensemble_config[key]
+                func = self.impl.get(config.getImplementationType())
+                if func is None:
+                    continue
+                data = func(config, realization_index)
+                if data is not None:
+                    group, dataset = data
+                    yield realization_index, group, dataset
+
+
+@contextmanager
+def chdir(path: str) -> None:
+    current = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(current)
+
+
+def dump_data(ert_config: str, case: str) -> None:
+    ert_config = os.path.join(os.path.dirname(__file__), "..", ert_config)
+    output_file = os.path.join(
+        os.path.dirname(__file__), f"{os.path.basename(os.path.dirname(ert_config))}.nc"
+    )
+    with chdir(os.path.dirname(ert_config)):
+        reader = Reader(ert_config, case)
+
+        mode = "w"
+        for realization_index, group, dataset in reader.load_all():
```

Comment:
Should we do `tqdm(reader.load_all())` since this might take a while?

Reply:
I think a less complex codebase is preferable to performance, especially considering this script will ideally only be run once ever.